examples/utils/example_client.rs

```diff
@@ -63,20 +63,19 @@ impl ExampleClient {
                         break 'outer;
                     }
                     Event::Terminate => {
-                        println!("Client failed to connect to the network. Restarting.");
-                        thread::sleep(Duration::from_secs(5));
-                        break;
+                        println!("Client failed to connect to the network.");
+                        return None;
                     }
                     _ => (),
                 }
             }
         }
 
-        ExampleClient {
+        Some(ExampleClient {
             routing_client: routing_client,
             receiver: receiver,
             full_id: full_id,
-        }
+        })
```

Comment:
Are those changes (returning an option everywhere) related to the disjoint groups implementation?
Otherwise I'd not do that now.